### DISCUSSION TITLE
automatic updating of plugins inside MA3 onPC when *.lua file changes

### DISCUSSION THREAD

**Discussion by lukas-runge** [hossimo/GMA3Plugins#24] [Ideas]
Hey everyone,

I developed a lot today and thought: "Wouldn't it be cool if reimporting the plugin over and over again after every change wouldn't be necessary anymore?"

I got 3 ideas to realize this:
1. This is already in some way implemented into the console and I am just not able to configure it correctly e.g. what effect does the `LoadOnDemand` option have? Haven't really figured it out.
2. Run a "main event loop" with LUA every second to reimport every plugin in pool. (scales very badly with tons of plugins in pool)
3. Run a "main event loop" with LUA every second to watch for file changes and then only trigger reimport when files have actually changed. This may require two different approaches on onPC and the console as it uses host function to determine if a file has changed.

Is there another solution I haven't thought of? Has anybody already implemented one of those three solutions and has a recommendation if it's a good idea or not? **Is this even a cool solution for development or should we rather try to get the debugging tools working?** (see lib_plugins/examples/debug_example/debug_example.lua 😉)

**Thanks already for letting me bounce my idea!**
Kind regards
@lukas-runge

**Comment by hossimo** [comment]
I use a Macro similar to Import `Plugin 1 "XML_NAME" /NC` and manually load reload the plugin after I run edit the file.


I have gotten the debug tools working in the past. I actually have them commented out in my test plugin code that I use to test theories quickly. I honestly didn't play with it very much but I was able to hit a breakpoint.

``` lua
local function Main(display_handle,argument)
    -- require("mobdebug").start()
    -- require("mobdebug").on()
    Echo("DEBUG CODE");
    -- require("mobdebug").off()
end

```

**Reply by lukas-runge** [reply]
Would be interesting if we could figure out how vscode and grandMA3 as well as the development project have to be configured to make remote debugging work. That should improve the development process a lot. I have just tooled around a bit but did not manage to get it working yet. Maybe you are able to recreate and documenent how you got it working. 😍👌

Kind regards
@lukas-runge
